File: packages/posthog/ios/Plugin/Classes/Options/CaptureOptions.swift
import Foundation
import Capacitor

@objc public class CaptureOptions: NSObject {
    private var event: String
    private var properties: [String: Any]?

    init(event: String, properties: JSObject?) {
        self.event = event
        self.properties = PosthogHelper.createHashMapFromJSObject(properties)
    }

    func getEvent() -> String {
        return event
    }

    func getProperties() -> [String: Any]? {
        return properties
    }
}

File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/options/CaptureOptions.java
package io.capawesome.capacitorjs.plugins.posthog.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.getcapacitor.JSObject;

import org.json.JSONException;

import java.util.Map;

import io.capawesome.capacitorjs.plugins.posthog.PosthogHelper;

public class CaptureOptions {
    @NonNull
    private String event;

    @Nullable
    private Map<String, Object> properties;

    public CaptureOptions(@NonNull String event, @Nullable JSObject properties) throws JSONException {
        this.event = event;
        this.properties = PosthogHelper.createHashMapFromJSONObject(properties);
    }

    @NonNull
    public String getEvent() {
        return event;
    }

    @Nullable
    public Map<String, Object> getProperties() {
        return properties;
    }
}
